site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -2365,6 +2365,78 @@ public function saveGradedComponent(TaGradedGradeable $ta_graded_gradeable, Grad
         }
     }
 
+    /**
+     * Route for saving the marks the submitter received for a component
+     */
+    #[Route("/courses/{_semester}/{_course}/gradeable/{gradeable_id}/grading/graded_gradeable/change_grade_version", methods: ["POST"])]
+    public function ajaxUpdateGradedVersionForStudent(string $gradeable_id): ?Jsonresponse {
```

Comment:
A route should never return null.  A proper response should always be returned.  Also, it looks like you have a typo in the return type name.

Reply:
Other routes in the same file also return `null` in the same situation, and also seem to be user-facing routes (like `ajaxSetPeerFeedback`), so I assumed it was a special case or something.

Reply:
Resolved with https://github.com/Submitty/Submitty/pull/10679/commits/db4c3dc26fbb4f4c089c58fc7d586238593afdc8